tests/storage/test_profile.py

```diff
@@ -132,3 +125,63 @@ def f(txn: LoggingTransaction) -> None:
         )
         self.assertEqual(len(res), len(expected_values))
         self.assertEqual(res, expected_values)
+
+    def test_upgrade_function(self) -> None:
```

Comment:
I don't know if this is necessary/should stay - I added it mostly for completeness/to show that I had tested the upgrade. Same for the other upgrade test in `test_user_filters`.

Reply:
This test (and the other one I mentioned) is the cause of the two test failures as the path to load the module to test the upgrade is different for the two failing tests (old-deps and building the release artifacts) - thinking about it I don't think that it makes sense to leave these tests in as the upgrades will run once so we don't have to worry about future changes breaking them - but I did want to make it clear that I did test the upgrade. Thoughts appreciated!

Reply:
I think testing of our migrations (especially our Python ones) is a fabulous idea...that being said, we don't usually do it.

Ideas that could make this easier to test:

* Allow the test file to somehow "pause" schema deltas and run some code (i.e. to insert users before your deltas are run) and then let it continue setting up the database.
* Move the code *out* of the schema delta and import it both in your tests and in the schema delta.
* Remove them. 🤷

Reply:
So my preference would be to remove the tests - my thinking is that they have run successfully in CI (like in the Trial python 3.7 run) so they have served their purpose to that point, and since the upgrade will only be run once, it's not something that we need a test around for as it's not something that someone will likely change in the future.